src/rebar_agent.erl

```diff
@@ -131,40 +131,98 @@ maybe_show_warning(State) ->
 %% that makes sense.
 -spec refresh_paths(rebar_state:t()) -> ok.
 refresh_paths(RState) ->
-    ToRefresh = (rebar_state:code_paths(RState, all_deps)
-                 ++ [filename:join([rebar_app_info:out_dir(App), "test"])
-                     || App <- rebar_state:project_apps(RState)]
-                %% make sure to never reload self; halt()s the VM
-                ) -- [filename:dirname(code:which(?MODULE))],
+    RefreshPaths = lists:usort(
+        application:get_env(rebar, refresh_paths, [])
```

Comment:
It would be a better idea (I think) to set this default value to `[all_deps, test]`. This would let us keep the current functionality, but avoid having to support magical values `no_deps` and `no_test` -- instead, just setting the values would make things work

Reply:
If set this default value to `[all_deps, test]`.But when I only need one, I have to know what other options are. There are only two options that are not a problem now, but there may be more than two in the future(I think). So here I am designing a subtraction option, just like `debug_info` and `no_debug_info`.

Reply:
Yes, I understand how it works, the problem with the format is that if other values were to be used, we need to create their opposite every time. The format of `debug_info` and `no_debug_info` is a bad workaround because there is no way to set `{debug_info, true|false}`, which instead is a much better format for configuration.